### DISCUSSION TITLE
Welcome to Nashpy Discussions!

### DISCUSSION THREAD

**Discussion by drvinceknight** [drvinceknight/Nashpy#92] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by clarkpetri** [comment]
Dr. Knight,


Apologies for an inevitably basic question but I'm slamming my head into the wall with a run error when importing nashpy. I have the most recent versions of pandas, lumpy, and scipy, as well as a fresh 'pip install nashpy'

When running a file with nothing more than 'import nashpy as nash' from the terminal I get the following error:

AttributeError: module 'numpy.typing' has no attribute 'NDArray'

I'm on an M1 Mac, anaconda, all other packages for analytical purposes (Pyomo, numpy, pandas, networkx, etc) are working fine. Thoughts? Thank you for such a great tool.

Clark

**Reply by drvinceknight** [reply]
Hi @clarkpetri, no reason to apologise (and sorry for getting back to you with a delay). A lot of my students are going through the same issue.

This is due to the recent addition of type hints throughout the code base. When seeing this on my students' machines, it seems to be due to the version of `numpy` not being completely up to date but as far up to date as the version of Anaconda you're running will allow.

One fix [that I have not tried] is:

```
$ conda update
```

and then potentially updating `numpy`.

Another option available to you is to create a virtual env for game theory (this I know works):

```
$ conda new --name gt numpy scipy matplotlib
```

that will create a new python environment with a number of dependencies.

Then:

```
$ conda activate gt
$ python -m pip install nashpy
```

which installs `nashpy` in the new environment.

Then you should be good to go.

Let me know if that doesn't fix your issue. Feel free to get in touch on [discord](https://discord.gg/wZsv6s8TTB) or [gitter](https://gitter.im/drvinceknight/Nashpy?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge).

**Reply by clarkpetri** [reply]
Dr. Knight,

Thank you for such a comprehensive reply. It certainly sounds like I was facing the NumPy version limitation based on Anaconda. I ended up thoroughly uninstalling Anaconda because back during my MS I had issues with it and now in my PhD program I've been experiencing them again. 

A brew installation with clean pyenv management between the default MacOS 2.7 and 3.10.2 solved the problem. Now I have nashpy running. Thanks again!

Clark Petri

**Reply by drvinceknight** [reply]
That's great, please keep in touch, I'm always glad to hear what people are using `nashpy` for :+1: 

Best of luck with the PhD 👍

**Comment by ThanosKa** [comment]
Hi, I would like to ask if anybody had the same issue as mine.

In jupyter NB  nashpy runs normally without any problem.
The problem is when I want to run the program as .py in VSC I have the error
"ModuleNotFoundError: No module named 'numpy'" and when i  pip install it i have the msg: "Requirement already satisfied: nashpy in /Users/admin/opt/anaconda3/lib/python3.9/site-packages (0.0.35)" 

Anyone can help? 

OS: Mac OS (M1)

**Reply by drvinceknight** [reply]
Hi @ThanosKa, this is usually due to the fact that the python interpreted your VSC is using is not the same as the one you installed Nashpy in to. I'm not a VSC user myself but @11michalis11 might be able to point you at something that helps you sort this out.

**Reply by MichalisPanayides** [reply]
Hello @ThanosKa. If you are using a virtual environment for your project, maybe the python kernel you are using in the Jupyter notebook does not match the python interpreter you are using to run the `.py` file. 


# To check the notebook kernel you are using:
- If you are using VS code to run jupyter notebooks, on the upper right corner you can change between kernels:

![image](https://user-images.githubusercontent.com/20097997/207916263-540ad672-9b27-4e7a-b41f-685aeab78a79.png)


# To check your python interpreter:
Find the `Select Interpreter ` command from the `Command Palette` and make sure it matches your jupyter notebook kernel.

`View -> Command Palette` and then type `Python: Select interpreter` 

![image](https://user-images.githubusercontent.com/20097997/207919257-070502e1-19a5-4a97-a59a-fda03110dd22.png)
